### DISCUSSION TITLE
import `tailwind.config.cjs` file to `*.tsx` component in react

### DISCUSSION THREAD

**Discussion by wojtekantonik** [tailwindlabs/tailwindcss#9083] [Help]
Hello,
I set up vite+typescript+tailwind project following this instruction https://larainfo.com/blogs/install-setup-vite-react-typescript-tailwind-css-3. I want to import my config to component as is shown here https://tailwindcss.com/docs/configuration#referencing-in-java-script, I'm typing relative path to config and change extension to `cjs` and I'm getting following error (pic .1) (copy version: `react.development.js:1407 Uncaught SyntaxError: The requested module '/tailwind.config.cjs?t=1660292737461&import' does not provide an export named 'default' (at Modal.component.tsx:5:8)`). So here is my question is this an error that is on tailwind's side or vite's or typescript's or something else?

<img width="776" alt="Zrzut ekranu 2022-08-12 o 10 50 46" src="https://user-images.githubusercontent.com/105207254/184319827-8571e39b-5b13-4c29-90b8-a39584e816f1.png">

**Comment by ptrxyz** [comment]
Try to add your `tailwind.config.cjs` to the following settings of your `vite.config.js`:
`optimizeDeps.include`, `build.commonjsOptions` and `resolve.alias`

This should make Vite convert it to an ESM module before it's loaded.
